pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -378,7 +463,12 @@ public void replaceLLSegment(String segmentName, IndexLoadingConfig indexLoading
     try {
       File indexDir = new File(_indexDir, segmentName);
       Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, _tableNameWithType);
-      addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, schema));
+      PartitionUpsertMetadataManager partitionUpsertMetadataManager = null;
+      if (_tableUpsertMetadataManager != null) {
+        int partitionId = new LLCSegmentName(segmentName).getPartitionId();
```

Comment:
We should find a better way of getting `partitionId` than from the segment name. We are looking to integrate with other streams where partition ID is not an integer (e.g. kinesis). In that case, this will not work.
@npawar may also want to comment on this

Reply:
as of now I don't see any other way of getting the partition id. Separately, we might want to start changes to put partition id/ sequence number into the segment zk metadata.